integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,398 @@
+            String whereClause = DataQualityConstants.PROCESSING_DTTM_COLUMN + " = '" + processing_dttm + "'";
+
+            setSourceRowCount();
+
+            long rowCount;
+            rowCount = getRowCount(databaseName, feedTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_FEED_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            rowCount = getRowCount(databaseName, invalidTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_INVALID_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            rowCount = getRowCount(databaseName, validTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_VALID_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            // Execute Data Quality rules
+            boolean rulePass;
+            boolean dqAllRulePass = true;
+            for (DataQualityRule rule : ruleList) {
+
+                log.info("Executing rule: " + rule.getName());
+
+                if (rule.loadAttributes(flowAttributes)) {
+                    rulePass = rule.evaluate();
+                } else {
+                    rulePass = false;
+                }
+
+                if (!rulePass) {
+                    log.error("Rule: " + rule.getName() + " has FAILED. Rule desc: " + rule.getDescription());
+                } else {
+                    log.info("Rule: " + rule.getName() + " has PASSED");
+                }
+
+                dqAllRulePass = dqAllRulePass && rulePass;
+
+            }
+
+            outputToHive();
+
+            outputToLog();
+
+            // Return false if any rule failed
+            if (dqAllRulePass == false) {
+                log.warn("Data Quality check failures");
+                isSuccessful = false;
+            } else {
+                isSuccessful = true;
+            }
+
+        } catch (MissingAttributeException e) {
+            log.error("Required Attribute missing from passed in data");
+            isSuccessful = false;
+        } catch (Exception e) {
+            log.error("Generic exception in doQualityCheck()", e);
+            isSuccessful = false;
+        }
+
+        return isSuccessful;
+    }
+
+    /**
+     * This method will set the source row count if it does not already exists. This happens when
+     * other processors are used to pull data, such as ImportSqoop
+     * 
+     * If ImportSqoop Nifi processor is used, it updates the source.row.count value with the sqoop
+     * count
+     */
+    protected void setSourceRowCount() {
+        try {
+            if (flowAttributes.containsAttribute(DataQualityConstants.SQOOP_ROW_COUNT_ATTRIBUTE)) {
+                String sqoopRowCount = flowAttributes.getAttributeValue(DataQualityConstants.SQOOP_ROW_COUNT_ATTRIBUTE);
+                flowAttributes.addAttribute(DataQualityConstants.SOURCE_ROW_COUNT_ATTRIBUTE, sqoopRowCount);
+            }
+        } catch (MissingAttributeException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)